### PR TITLE
[#381] Add `prepare` hook for action-button

### PR DESCRIPTION
**Pull request by solomonhawk**
**What kind of change does this PR introduce?**

- Bugfix
- [x] Feature
- Code style update
- Refactor
- Build tools
- Documentation
- Other, please describe:

**Does this PR introduce a breaking change?**

- Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**Does this PR fulfill these requirements:**

- [x] All tests are passing
- [x] `yarn pretty` has been run
- [x] Commits reference open issues

**Additional Information**

See #381

### CODE REVIEW

File: src/addons/action-button.js [hunk 1]
```diff
@@ -24,11 +24,13 @@ class ActionButton extends React.PureComponent {
...
 
   static defaultProps = {
-    tag: 'button'
+    tag: 'button',
+    prepare: (event, value) => value
   }
 
   click(event: Event): Action {
-    let action = this.send(this.props.action, this.props.value)
+    let payload = this.props.prepare(event, this.props.value)
```
Comment (solomonhawk) [inline src/addons/action-button.js]: Any reason these two arguments should be reversed?

Comment (nhunzaker) [inline src/addons/action-button.js]: ActionForm's prepare is `(value) => value)`. For consistency, what do you think of doing `(value, event) => value` for both?

Comment (solomonhawk) [inline src/addons/action-button.js]: Makes sense to me. Willfix.

**Comment by solomonhawk** [comment]
@nhunzaker Merge when ready!

**Pull request merged**